index.html

```diff
@@ -296,141 +292,162 @@
+      scheduleObj.teacher = "";
+    }
+    if (scheduleObj.room == "N/A") {
+      scheduleObj.room = "";
+    }
+    // Add in photos if the user's schedule is being rendered
+    if (type == "full") {
+      // Add the left-hand images to the schedule
+      addScheduleImages(scheduleObj, schedule);
+    } else if (type == "lite") {
+      if (letterToNum(scheduleObj.period) <= 8) { // If it's a class period
+        // Determine if the class is shared
+        scheduleObj.shared = isSharedClass(scheduleObj);
+      } else {
+        return null; // Skip over lunch, advisory, assembly, etc.
+      }
+    }
+    // Split the time into the start and end times
+    var times = days[0][day][school][currentSlot]["time"].split('-');
+    scheduleObj.startTime = times[0].trim();
+    scheduleObj.endTime = times[1].trim();
+    return scheduleObj;
+  }
 
-          // If the class is a normal class but the user is a teacher
-          } else {
-            sanitized_teacher_name = schedule.firstname + "_" + schedule.lastname;
-            sanitized_teacher_name = sanitized_teacher_name.toLowerCase();
-            scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
-          }
-          scheduleObj.roomLink = "/room/" + scheduleObj.room.toLowerCase().replace(/ /g,"_");
+  // Add the images on the left side to full schedules
+  function addScheduleImages(scheduleObj, userSchedule) {
+    // If the user is a normal user and the class is a normal class
+    if (scheduleObj.teacher != "" && userSchedule.grade) {
+      var sanitized_teacher_name;
+      sanitized_teacher_name = scheduleObj.teacher.toLowerCase();
+      sanitized_teacher_name = sanitized_teacher_name.replace(/ /g,"_");
+      scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
+      scheduleObj.teacherLink = "/teacher/" + sanitized_teacher_name;
 
+    // If class is a special class or free period
+    } else if (scheduleObj.period == "X" || scheduleObj.name == "Free Period") {
+      scheduleObj.teacherLink = "";
+      var image_name = scheduleObj.name.toLowerCase();
+      image_name = image_name.replace("/", "");
+      image_name = image_name.replace(/\s/g, '');
+      scheduleObj.avatar = "/images/" + image_name + ".jpg";
 
-        } else if (type == "lite") {
-          if (letterToNum(scheduleObj.period) <= 8) { // If it's a class period
-            scheduleObj.shared = false; // Find out if the user has a class that period
-            for (var i = 0; i < userSchedule.classes.length; i++) {
-              if (userSchedule.classes[i].period == scheduleObj.period &&
-                  userSchedule.classes[i].name == scheduleObj.name) {
-                scheduleObj.shared = true;
-                break;
-              }
-            }
-          } else {
-            continue; // Skip over lunch, advisory, assembly, etc.
-          }
-        }
+    // If the class is a normal class but the user is a teacher
+    } else {
```

Comment:
Well, if they have a schedule and are not a student, I assume that their schedule should always be displayed like a teacher's.